xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -157,6 +157,59 @@ def match_and_rewrite(self, op: memref.LoadOp, rewriter: PatternRewriter, /):
         rewriter.replace_matched_op(ops, new_results=[load_result.res])
 
 
+@dataclass
+class ConvertSubviewOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref.SubviewOp, rewriter: PatternRewriter, /):
+        assert isinstance(op_memref_type := op.source.type, memref.MemRefType)
+
+        memref_type = cast(memref.MemRefType[Any], op_memref_type)
+        static_offsets = cast(tuple[int, ...], op.static_offsets.get_values())
+        offsets = cast(list[SSAValue], [])
+
+        for idx, offset in enumerate(static_offsets):
+            if offset == memref.SubviewOp.DYNAMIC_INDEX:
+                offsets.append(op.offsets[idx])
+            else:
+                rewriter.insert_op_before_matched_op(
+                    const_op := arith.ConstantOp(
+                        builtin.IntegerAttr(offset, builtin.IndexType())
+                    )
+                )
+                offsets.append(const_op.result)
+
+        # We can treat a subview as getting a pointer to the first element in the subview.
+        ops, target_ptr = get_target_ptr(op.source, memref_type, offsets)
+
+        rewriter.replace_matched_op(
+            (
+                *ops,
+                cast_op := builtin.UnrealizedConversionCastOp.get(
+                    [target_ptr], [op.result.type]
+                ),
+            )
+        )
+        op.result.replace_by(cast_op.results[0])
```

Comment:
Why is this necessary? The default behaviour should replace the result with the result of the last operation.